src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs

```diff
@@ -0,0 +1,256 @@
+using Newtonsoft.Json.Linq;
+using Promitor.Agents.ResourceDiscovery.Configuration;
+using Promitor.Agents.ResourceDiscovery.Graph.Interfaces;
+using Promitor.Agents.ResourceDiscovery.Graph.Model;
+using Promitor.Agents.ResourceDiscovery.Graph.Repositories;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Promitor.Tests.Unit.Agents.ResourceDiscovery
+{
+    public class AzureResourceRepositoryTests
+    {
+        private const string basePath = @"Agents\ResourceDiscovery\TestData\";
+
+        private readonly Mock<IAzureResourceGraph> azureResourceGraph;
+        private readonly Mock<IOptionsMonitor<ResourceDeclaration>> resourceDeclarationMonitor;
+        private readonly Mock<ILogger<AzureResourceRepository>> logger;
+
+        private readonly AzureResourceRepository azureResourceRepository;
+
+        public AzureResourceRepositoryTests()
+        {
+            azureResourceGraph = new Mock<IAzureResourceGraph>(MockBehavior.Strict);
+            resourceDeclarationMonitor = new Mock<IOptionsMonitor<ResourceDeclaration>>(MockBehavior.Strict);
+            logger = new Mock<ILogger<AzureResourceRepository>>(MockBehavior.Strict);
+
+            azureResourceRepository = new AzureResourceRepository(azureResourceGraph.Object, resourceDeclarationMonitor.Object, logger.Object);
+        }
+
+        [Fact]
+        public async Task AzureSubscriptions_Discovery_Succeeds()
+        {
+            var testData = ReadTestData("DiscoverAzureSubscriptionsAsync.json");
+            azureResourceGraph.Setup(m => m.QueryAzureLandscapeAsync("Discover Azure Subscriptions", It.IsAny<string>())).ReturnsAsync(testData);
+            var expectedSubscriptions = GetAzureSubscriptions();
+
+            var azureSubscriptions = await azureResourceRepository.DiscoverAzureSubscriptionsAsync();
+
+            Assert.NotStrictEqual(expectedSubscriptions, azureSubscriptions);
+
+            azureResourceGraph.Verify(m => m.QueryAzureLandscapeAsync("Discover Azure Subscriptions", It.IsAny<string>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task AzureResourceGroups_Discovery_Succeeds()
+        {
+            var testData = ReadTestData("DiscoverAzureResourceGroupsAsync.json");
+            azureResourceGraph.Setup(m => m.QueryAzureLandscapeAsync("Discover Azure Resource Groups", It.IsAny<string>())).ReturnsAsync(testData);
+            var expectedResourceGroups = GetAzureResourceGroups();
+
+            var azureResourceGroups = await azureResourceRepository.DiscoverAzureResourceGroupsAsync();
+
+            Assert.NotStrictEqual(expectedResourceGroups, azureResourceGroups);
+
+            azureResourceGraph.Verify(m => m.QueryAzureLandscapeAsync("Discover Azure Resource Groups", It.IsAny<string>()), Times.Once);
+        }
+
+        private List<AzureSubscriptionInformation> GetAzureSubscriptions()
+        {
+            return new List<AzureSubscriptionInformation> {
```

Comment:
Is this based on the test data JSONs? If so, let's add a quick comment to clarify this

Reply:
Sure